components/tag-list/tag-list.js

```diff
@@ -352,14 +353,17 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 		});
 	}
 
-	async _toggleHiddenTagVisibility() {
+	async _toggleHiddenTagVisibility(e) {
 		this._showHiddenTags = !this._showHiddenTags;
 
 		if (!this.shadowRoot) return;
 
 		await this.updateComplete;
-		const button = this.shadowRoot.querySelector('.d2l-tag-list-button');
-		if (button) button.focus();
+		if (e.target.classList.contains('d2l-tag-list-button-show-more')) this._items[this._chompIndex].focus();
```

Comment:
This switches the focus behaviour such that when the user selects the "+ X more" button, instead of their focus being taken to the "Show less" button, it is taken to the next list item that they wouldn't have seen/heard yet.